### PR TITLE
Add option to install plugin dependencies

### PR DESCRIPTION
**Pull request by dychen**
### Background

Currently, it is impossible to run plugins that require third party dependencies. For instance, the [Twitter](https://github.com/Significant-Gravitas/Auto-GPT-Plugins) plugin requires `tweepy` and `pandas`. Running Auto-GPT with this plugin errors out.

Reference thread: https://github.com/Significant-Gravitas/Auto-GPT/pull/757#issuecomment-1519187983

@ntindle 

### Changes

Updated the base CLI to optionally accept an `install-plugin-deps` parameter which installs plugin dependencies during setup. Temporarily extracts the top level `requirements.txt` file and runs `pip install` on it. Does not recursively search for `requirements.txt` files as there might be unintended behavior. Does this through a system call, which is probably not the best implementation in the long run.

Includes a new `install-plugin-deps` parameter which defaults to `false` to minimize interaction with existing code.

### Documentation

There is some documentation in the code.

### Test Plan

N/A

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by dychen** [comment]
@ntindle let me know if this is what you had in mind

File: autogpt/plugins.py [hunk 1]
```diff
@@ -18,6 +18,29 @@
 from autogpt.models.base_open_ai_plugin import BaseOpenAIPlugin
 
 
+def install_plugin_dependencies(zip_path: str, debug: bool = False) -> None:
+    """
+    Inspect a zipfile for a requirements file and install if exists. Only
+    searches at top level for now.
```
Comment (ntindle) [inline autogpt/plugins.py]: Remove “for now”

File: autogpt/plugins.py [hunk 2]
```diff
@@ -18,6 +18,29 @@
...
+
+    Returns:
+        None
+    """
+    try:
+        with zipfile.ZipFile(zip_path, "r") as zfile:
+            basedir = zfile.namelist()[0]
+            basereqs = os.path.join(basedir, 'requirements.txt')
+            extracted = zfile.extract(basereqs)
+            os.system(f"pip install -r {extracted}")
```
Comment (ntindle) [inline autogpt/plugins.py]: This is a potential pain point. Can you get the running python exec and use it with -m pip …

Comment (dychen) [reply autogpt/plugins.py]: Would you rather just have this all in `check_requirements.py`?

Comment (dychen) [reply autogpt/plugins.py]: https://github.com/Significant-Gravitas/Auto-GPT/pull/3068/commits/576f9a125e0afe3f4331d02a7d25001b017f561f

If this is better, I can revert the previous changes ^

**Comment by dychen** [comment]
@ntindle 

I have two options here:
1. Update `plugins.py` and the CLI interface. The dependencies are installed when the user runs: `python -m autogpt --install-plugin-deps`
2. Update `check_requirements.py`. The dependencies are installed alongside base project requirements automatically when the user runs: `run.sh` or `run.bat`

Let me know your preference. I think 2 might be better as it is minimally invasive.

**Comment by ntindle** [comment]
2 is preferred

**Comment by ntindle** [comment]
Did a few changes get dropped?

**Comment by ntindle** [comment]
I'd like this to exist within the CLI still but I'm going to test and merge this if it passes

**Review by ntindle — APPROVED** [review]

**Comment by dychen** [comment]
> I'd like this to exist within the CLI still but I'm going to test and merge this if it passes

My bad. Wasn't sure if you wanted it standalone, as part of `run.sh`/`check_requirements.py`, or as part of the CLI, so I opted for the least invasive path. I added it back in [this](https://github.com/dychen/Auto-GPT/commit/296155cf4179676247bcdce1809b3cbf1c8b6a58) commit, so the following should both work:

```
$ ./run.sh --install-plugin-deps
$ python -m autogpt --install-plugin-deps
```

Will open up a PR.

**Comment by ntindle** [comment]
Ping me when it’s up :)

**Pull request merged**